am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/ReturnRolePermissionsForCaseTypeIntegrationTest.java

```diff
@@ -162,8 +167,147 @@ void whenUnorderedListThenReturnOrderedList() {
+        importerService.addRole(idamRoleWithRoleBasedAccess1 = UUID.randomUUID().toString(), IDAM, PUBLIC, ROLE_BASED);
+        Set<Permission> permissions = ImmutableSet.of(UPDATE, DELETE);
+        Set<Permission> permissions1 = ImmutableSet.of(CREATE, READ);
+        importerService.addResourceDefinition(resourceDefinition1 =
+            createResourceDefinition(serviceName, resourceType, UUID.randomUUID().toString()));
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(idamRoleWithRoleBasedAccess,
+            resourceDefinition, "", permissions, PUBLIC));
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(idamRoleWithRoleBasedAccess1,
+            resourceDefinition1, "", permissions1, PUBLIC));
+
+        List caseTypeIds = ImmutableList.of(
+            UUID.randomUUID().toString(), UUID.randomUUID().toString());
+
+        List<RolePermissionsForCaseTypeEnvelope> result = importerService.getRolePermissionsForCaseType(
+            caseTypeIds);
+
+        assertThat(result.size()).isEqualTo(0);
+    }
+
+    @Test
+    void whenOneOfCaseTypeInListHasAttributeLevelPermissionsOnlyThenItIsNotReturned() {
+        String idamRoleWithRoleBasedAccess1;
+        ResourceDefinition resourceDefinition1;
+        importerService.addRole(idamRoleWithRoleBasedAccess1 = UUID.randomUUID().toString(), IDAM, PUBLIC, ROLE_BASED);
+        String resourceName1;
+        Set<Permission> permissions = ImmutableSet.of(UPDATE, DELETE);
+        Set<Permission> permissions1 = ImmutableSet.of(CREATE, READ);
+        importerService.addResourceDefinition(resourceDefinition1 =
+            createResourceDefinition(serviceName, resourceType, resourceName1 = UUID.randomUUID().toString()));
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(idamRoleWithRoleBasedAccess,
+            resourceDefinition, "/test", permissions, PUBLIC));
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(idamRoleWithRoleBasedAccess1,
+            resourceDefinition1, "", permissions1, PUBLIC));
+        List caseTypeIds = ImmutableList.of(
+            resourceDefinition.getResourceName(), resourceDefinition1.getResourceName());
+        List<RolePermissionsForCaseTypeEnvelope> result = importerService.getRolePermissionsForCaseType(
+            caseTypeIds);
+
+        //result.sort(comparing(RolePermissionsForCaseTypeEnvelope::getCaseTypeId));
+
+        assertThat(result.size()).isEqualTo(1);
+
+        List<RolePermissionsForCaseTypeEnvelope> expectedResourceAuditResult = ImmutableList.of(
+            RolePermissionsForCaseTypeEnvelope.builder().caseTypeId(resourceName1)
+                .defaultRolePermissions(ImmutableList.of(DefaultRolePermissions.builder()
+                    .role(idamRoleWithRoleBasedAccess1).permissions(permissions1).build())).build());
+
+        assertThat(result).isEqualTo(expectedResourceAuditResult.stream()
```

Comment:
may be .stream()
            .sorted(comparing(RolePermissionsForCaseTypeEnvelope::getCaseTypeId not required as excepted is single  
list

Reply:
👍